src/azure-cli/azure/cli/command_modules/acr/agentpool.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from msrest.exceptions import ValidationError
+from knack.log import get_logger
+from knack.util import CLIError
+from ._utils import (
+    get_registry_by_name,
+    validate_managed_registry
+)
+
+
+DEFAULT_COUNT = 1
+DEFAULT_TIER = 'S1'
+DEFAULT_OS = 'Linux'
+
+logger = get_logger(__name__)
+
+
+def acr_agentpool_create(cmd,
+                         client,
+                         agent_pool_name,
+                         registry_name,
+                         resource_group_name=None,
+                         count=DEFAULT_COUNT,
+                         tier=DEFAULT_TIER,
+                         os_type=DEFAULT_OS,
+                         vnet_id=None):
+
+    registry, resource_group_name = get_registry_by_name(
+        cmd.cli_ctx, registry_name, resource_group_name)
+
+    AgentPool = cmd.get_models('AgentPool')
+
+    agentpool_create_parameters = AgentPool(
+        location=registry.location,
+        count=count,
+        tier=tier.upper(),
+        os=os_type,
+        VirtualNetworkSubnetResourceId=vnet_id
+    )
+
+    try:
+        return client.create(resource_group_name=resource_group_name,
+                             registry_name=registry_name,
+                             agent_pool_name=agent_pool_name,
+                             agent_pool=agentpool_create_parameters)
+    except ValidationError as e:
+        raise CLIError(e)
+
+
+def acr_agentpool_update(cmd,
+                         client,
+                         agent_pool_name,
+                         registry_name,
+                         resource_group_name=None,
+                         count=None):
+
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name)
+
+    try:
+        return client.update(resource_group_name=resource_group_name,
+                             registry_name=registry_name,
+                             agent_pool_name=agent_pool_name,
+                             count=count)
+    except ValidationError as e:
+        raise CLIError(e)
+
+
+def acr_agentpool_delete(cmd,
+                         client,
+                         agent_pool_name,
+                         registry_name,
+                         resource_group_name=None):
+
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name)
+
+    try:
+        response = client.delete(resource_group_name=resource_group_name,
+                                 registry_name=registry_name,
+                                 agent_pool_name=agent_pool_name)
+        if response.status_code == 404:
+            response.status_code = 200
```

Comment:
could you pls share more context why 404 but map it to 200 for user?

Reply:
Yes there seems to be a timing issue between arm and our rp, and it sometimes return 404 when deleted. Although I'm changing the workarround please ignore this.

Reply:
I'm adding this comment for context in next solution:
/* Since agent pool is a tracked resource in arm,
                                 * arm, as well, pings the async deletion api at the
                                 * same time to get the status. If arm gets the 200 status
                                 * first and knows that the resource is deleted, it marks the
                                 * resource as deleted and stop routing further requests
                                 * to the resource including the async deletion status
                                 * api. Hence arm will directly return 404. Consider this as
                                 * successful delete
                                 */